src/operand/reference.ts

```diff
@@ -27,14 +42,29 @@ function contextValueLookup (ctx: Context, key: string): Result {
   let pointer = ctx
 
   for (let i = 0; i < keys.length; i++) {
+    let currentValue = pointer[keys[i]]
+
+    // Resolve array notation
+    const matchesArrayNotation = /^(.+)\[(\d+)\]$/.exec(keys[i])
```

Comment:
Will it handle an array of an array? e.g. `RefH[0][1]`

Reply:
That's a good point, probably not. I guess this needs to be a loop as well

Reply:
7f0434bc7fc616653450b2ed4d4885d306603d2d

Reply:
d4bb12e89809d55826168b69eb182bebef7ef84b